fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/limits/FDBRecordStoreByteLimitTest.java

```diff
@@ -0,0 +1,471 @@
+     * predicates get planned as {@link com.apple.foundationdb.record.query.plan.plans.RecordQueryUnorderedUnionPlan}s,
+     * which have unusual semantics where results are returned in an undefined order as soon as any child has one.
+     * Therefore, the assertions made in {@link #assertPlanLimitsWithCorrectExecution(List, FDBRecordContext, RecordQueryPlan)}
+     * are far too strong for plans like this. Instead, we make very weak assertions that the byte scan limit does
+     * <emph>something</emph>.
+     */
+    @ParameterizedTest
+    @MethodSource("complexTextQueries")
+    public void queryWithWideOrOfFullTextPrefixPredicates(@Nonnull RecordQuery query, int numPredicates) throws Exception {
+        deleteSimpleRecords();
+        final List<String> textSamples = ImmutableList.of(
+                TextSamples.ANGSTROM,
+                TextSamples.ROMEO_AND_JULIET_PROLOGUE,
+                TextSamples.AETHELRED,
+                TextSamples.FRENCH,
+                TextSamples.KOREAN
+        );
+
+        RecordMetaDataHook indexHook = metaDataBuilder ->
+                    metaDataBuilder.addIndex(metaDataBuilder.getRecordType(SIMPLE_DOC), SIMPLE_TEXT_PREFIX);
+        try (FDBRecordContext context = openContext()) {
+            openTextRecordStore(context, indexHook);
+            for (int i = 0; i < textSamples.size(); i++) {
+                recordStore.saveRecord(TestRecordsTextProto.SimpleDocument.newBuilder()
+                        .setDocId(i)
+                        .setGroup(i % 2)
+                        .setText(textSamples.get(i))
+                        .build());
+            }
+            commit(context);
+        }
+
+        setupPlanner(null);
+        RecordQueryPlan plan = planner.plan(query);
+        assertThat(plan, descendant(unorderedUnion(Collections.nCopies(numPredicates, any(RecordQueryPlan.class)))));
+
+        long totalBytes;
+        try (FDBRecordContext context = openContext()) {
+            openTextRecordStore(context, indexHook);
+            context.getTimer().reset();
+            recordStore.executeQuery(query, null, ExecuteProperties.SERIAL_EXECUTE).asList().get();
+            totalBytes = byteCounter.getBytesScanned(context);
+        }
+
+        try (FDBRecordContext context = openContext()) {
+            openTextRecordStore(context);
+            context.getTimer().reset();
+            final ExecuteProperties executeProperties = ExecuteProperties.newBuilder()
+                    .setScannedBytesLimit(0)
+                    .build();
+            RecordCursor<?> cursor = recordStore.executeQuery(query, null, executeProperties);
+            RecordCursorResult<?> result;
+            do {
+                result = cursor.onNext().get();
+            } while (result.hasNext());
+            assertThat(byteCounter.getBytesScanned(context), lessThan(totalBytes));
+            assertEquals(RecordCursor.NoNextReason.BYTE_LIMIT_REACHED, result.getNoNextReason());
```

Comment:
In theory, this could still assert that the *set* of results is the same regardless of the byte limit (across continuations). The *bag* of results isn't guaranteed, though (because of extra duplicates).